packages/cli/src/services/community.service.ts

```diff
@@ -0,0 +1,306 @@
+import { exec } from 'child_process';
+import { access as fsAccess, mkdir as fsMkdir } from 'fs/promises';
+
+import { LoggerProxy as Logger } from 'n8n-workflow';
+import { UserSettings } from 'n8n-core';
+import { Service } from 'typedi';
+import { promisify } from 'util';
+import axios from 'axios';
+
+import config from '@/config';
+import { toError } from '@/utils';
+import { InstalledPackagesRepository } from '@/databases/repositories/installedPackages.repository';
+import type { InstalledPackages } from '@/databases/entities/InstalledPackages';
+import {
+	NODE_PACKAGE_PREFIX,
+	NPM_COMMAND_TOKENS,
+	NPM_PACKAGE_STATUS_GOOD,
+	RESPONSE_ERROR_MESSAGES,
+	UNKNOWN_FAILURE_REASON,
+} from '@/constants';
+
+import type { PublicInstalledPackage } from 'n8n-workflow';
+import type { PackageDirectoryLoader } from 'n8n-core';
+import type { CommunityPackages } from '@/Interfaces';
+import type { LoadNodesAndCredentials } from '@/LoadNodesAndCredentials';
+
+const {
+	PACKAGE_NAME_NOT_PROVIDED,
+	DISK_IS_FULL,
+	PACKAGE_FAILED_TO_INSTALL,
+	PACKAGE_VERSION_NOT_FOUND,
+	PACKAGE_NOT_FOUND,
+} = RESPONSE_ERROR_MESSAGES;
+
+const {
+	NPM_PACKAGE_NOT_FOUND_ERROR,
+	NPM_NO_VERSION_AVAILABLE,
+	NPM_DISK_NO_SPACE,
+	NPM_DISK_INSUFFICIENT_SPACE,
+	NPM_PACKAGE_VERSION_NOT_FOUND_ERROR,
+} = NPM_COMMAND_TOKENS;
+
+const asyncExec = promisify(exec);
+
+const INVALID_OR_SUSPICIOUS_PACKAGE_NAME = /[^0-9a-z@\-./]/;
+
+@Service()
+export class CommunityService {
```

Comment:
Should we call this `CommunityNodesService` instead ?

Reply:
If we want to be strict, the reference might best be to "community packages", which contain both nodes and creds. I went with `CommunityService` to keep it short and generic. Let me know!

Reply:
maybe `CommunityPackagesService` then?

Reply:
I kind of liked the conciseness of `CommunityService`. Went with `CommunityPackageService` in singular for consistency.